lib/src/Uncertainty/Bayesian/Gibbs.cxx

```diff
@@ -134,6 +197,28 @@ Gibbs::MetropolisHastingsCollection Gibbs::getMetropolisHastingsCollection() con
   return samplers_;
 }
 
+/* Is order random accessors */
+void Gibbs::setUpdatingMethod(const UpdatingMethod updatingMethod)
+{
+  if (updatingMethod > 1)
+    throw InvalidArgumentException(HERE) << "Updating method should be 0 (DETERMINISTIC_UPDATING) or 1 (RANDOM_UPDATING)";
+  if (updatingMethod != updatingMethod_) // reset
+  {
+    samplesNumber_ = 0;
+    previouslyChosenSampler_ = samplers_.getSize(); // no sampler has that index
+  }
+  updatingMethod_ = updatingMethod;
+}
+
+Gibbs::UpdatingMethod Gibbs::getUpdatingMethod() const
+{
+  if (updatingMethod_ == 0)
+    return UpdatingMethod::DETERMINISTIC_UPDATING;
+  if (updatingMethod_ == 1)
+    return UpdatingMethod::RANDOM_UPDATING;
+  throw InvalidArgumentException(HERE) << "Updating method should be 0 (DETERMINISTIC_UPDATING) or 1 (RANDOM_UPDATING)";
```

Comment:
I dont think there's a point in checking arguments in the getter, only in the setter

Reply:
How do you do that? Like this ?
```
 return UpdatingMethod(updatingMethod_)
```

Reply:
maybe with `static_cast<UpdatingMethod>(updatingMethod_)`

Reply:
Ok, I have committed your suggestion. Let's see if it works.